src/Sulu/Bundle/WebsiteBundle/Command/DumpSitemapCommand.php

```diff
@@ -120,16 +135,33 @@ private function dumpSitemap(PortalInformation $portalInformation)
             $portalInformation->setUrl(str_replace('{host}', $this->defaultHost, $portalInformation->getUrl()));
         }
 
-        if (!$this->sitemapProviderPool->needsIndex()) {
+        $filePath = $this->sitemapRenderer->getIndexDumpPath(
+            $this->scheme,
+            $portalInformation->getWebspaceKey(),
+            $portalInformation->getLocale(),
+            $portalInformation->getHost()
```

Comment:
if we want to reuse this function in the controller we have to pass it individually. In the controller we cannot use the portal.